### PR TITLE
Add explicit disequality handling when generating side condition for CBQI BV.

### PR DESCRIPTION
**Pull request by aniemetz**
This refactors solveBvLit to support explicit handling of disequalities (and, in the next step, inequalities) when generating side conditions.

### CODE REVIEW

**Comment by aniemetz** [comment]
`pol` and `litk` are updated accordingly as soon as a choice term is generated: https://github.com/CVC4/CVC4/pull/1447/files#diff-539f588549cb794cd2e3f954deac343dR1107
I decided to not do this in the getScBv* functions since we maintain the invariant that as soon as we generate a side condition, `pol` and/or `litk` gets flipped to `true` and `EQUAL`. Doing it were it is done right now avoids unnecessary code duplication.

**Comment by ajreynol** [comment]
Ah yeah, you're right. The diff is pretty confusing for this PR unfortunately, I couldn't follow. 

So I'm wondering then if that code should always be applied?  Isn't it the case that the PLUS inversion for > keeps >?  In other words we should be doing:
```
x+s>t
x>t-s
x = (choice k. k>(t-s))
```
whereas would the current code do this?:
```
x+s>t
x=t-s
```

**Comment by ajreynol** [comment]
Replying to my own comment, I can see that the code is correct since we handle the easy inversions separately.
Anyways, my more general concern is that some of the operators in the main switch statement would want to do the conversion in different ways. But perhaps this isn't a concern.

**Comment by aniemetz** [comment]
The conversion is only done for the operators where we generate side conditions and therefore have to flip `pol = true` and `litk = EQUAL`. So, I think this isn't a concern.

And yes, for `x + s < t` we keep `litk = <` until we either generate a side condition for some other operator or traversed all the way down, where we then generate the side `x = (choice k. t-s != 0 => k<(t-s))` (https://github.com/CVC4/CVC4/pull/1447/files#diff-539f588549cb794cd2e3f954deac343dR1116).

**Comment by aniemetz** [comment]
> The conversion is only done for the operators where we generate side conditions and therefore have to flip pol = true and litk = EQUAL. So, I think this isn't a concern.

I just realized that this might be a little hard to understand.

What I meant is that all the operators that generate side conditions are handled in the main switch statement, or rather, all operators handled in the main switch statement generate side conditions. I explicitly separated these operators from the operators that do not generate side conditions (PLUS, SUB, XOR) for exactly that purpose (i.e., uniform handling of operators that generate and don't generate side conditions). Hence, this shouldn't be a concern.

**Comment by ajreynol** [comment]
Alright, we will reorganize as we go if we hit more cases that are not uniform, this looks fine to me.

File: src/theory/quantifiers/bv_inverter.cpp [hunk 1]
```diff
@@ -288,342 +295,547 @@ static Node getScBvSlt(bool pol, Kind k, unsigned idx, Node x, Node t)
...
   NodeManager* nm = NodeManager::currentNM();
+  Node scl, scr;
+
+  if (idx == 0)
+  {
+    Assert (pol == false);
+    /* x % s != t
+     * with side condition:
+     * s != 1 || t != 0  */
+    unsigned w = bv::utils::getSize(s);
```
Comment (mpreiner) [inline src/theory/quantifiers/bv_inverter.cpp]: Move `w` before if since you need the bit width in every branch.

**Pull request merged**